pkg/userdata/helper/kubelet.go

```diff
@@ -279,8 +297,33 @@ func kubeletConfiguration(clusterDomain string, clusterDNS []net.IP, featureGate
 }
 
 // KubeletFlags returns the kubelet flags.
-func KubeletFlags(version, cloudProvider, hostname string, dnsIPs []net.IP, external bool, pauseImage string, initialTaints []corev1.Taint, extraKubeletFlags []string) (string, error) {
-	tmpl, err := template.New("kubelet-flags").Funcs(TxtFuncMap()).Parse(kubeletFlagsTpl)
+func KubeletFlags(version, cloudProvider, hostname string, dnsIPs []net.IP, external bool, ipFamily util.IPFamily, pauseImage string, initialTaints []corev1.Taint, extraKubeletFlags []string) (string, error) {
+	// --node-ip and --cloud-provider kubelet flags conflict in the dualstack setup.
+	// In general, it is not expected to need to use --node-ip with external CCMs,
+	// as the cloud provider is expected to know the correct IPs to return.
+
+	// For details read kubernetes/sig-networking channel discussion
+	// https://kubernetes.slack.com/archives/C09QYUH5W/p1654003958331739
+
+	withCloudProvider := true
```

Comment:
Ok so this parameter means, whether a CCM is supported or not, can you please have a better name for it? as `withCloudProvider` is super vague and doesn't really have an obvious meaning in this context. maybe something like `ccmSupported` or whatever naming that reflects the functionality of this guy :)

Reply:
Oh and small nit, in go usually default values for bools is false, so that you can have early exit of the code

Reply:
It is about kubelet's `--cloud-provider` flag. Changed name to make that clear.

Reply:
> Oh and small nit, in go usually default values for bools is false...

Here the value will always be true except in one case. So makes sense to start with true.